kubernetes/samples/charts/traefik/README.md

```diff
@@ -4,102 +4,107 @@ This sample demonstrates how to install the Traefik ingress controller to provid
+First, you need to create two secrets with TLS certificates, one with the common name `domain1.org`, the other with the common name `domain2.org`. We use `openssl` to generate self-signed certificates for demonstration purposes. Note that the TLS secret needs to be in the same namespace as the WebLogic domain.
 ```
 # create a TLS secret for domain1
 $ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls1.key -out /tmp/tls1.crt -subj "/CN=domain1.org"
-$ kubectl create secret tls domain1-tls-cert --key /tmp/tls1.key --cert /tmp/tls1.crt
-
+$ kubectl -n weblogic-domain1 create secret tls domain1-tls-cert --key /tmp/tls1.key --cert /tmp/tls1.crt
 # create a TLS secret for domain2
 $ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls2.key -out /tmp/tls2.crt -subj "/CN=domain2.org"
-$ kubectl -n test1 create secret tls domain2-tls-cert --key /tmp/tls2.key --cert /tmp/tls2.crt
-```
-Then deploy the TLS Ingress.
-```
+$ kubectl -n weblogic-domain2 create secret tls domain2-tls-cert --key /tmp/tls2.key --cert /tmp/tls2.crt
+
+# Deploy a TLS IngressRoute.
 $ kubectl create -f samples/tls.yaml
+ingressroute.traefik.containo.us/traefik-tls-1 created
+ingressroute.traefik.containo.us/traefik-tls-2 created
 ```
-Now you can access the two WLS domains with different hostnames using the HTTPS endpoint.
+Now you can access the application on the WebLogic domain with the hostname in HTTP header.
+The ingress controller secure port can be obtained dynamically from the `traefik-operator` service in the `traefik` namespace.
 ```
-$ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30443/testwebapp/
-$ curl -k -H 'host: domain2.org' https://${HOSTNAME}:30443/testwebapp/
+# Get the ingress controller secure web port
+TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.ports[?(@.name=="websecure")].nodePort}'`
+$ curl -k -H 'host: domain1.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
 ```
 
 ## Uninstall the Traefik operator
 After removing all the Ingress resources, uninstall the Traefik operator:
 ```
-$ helm delete --purge traefik-operator
+$ helm uninstall traefik-operator --namespace traefik --keep-history
 ```
-## Install and uninstall the Traefik operator with setup.sh
-Alternatively, you can run the helper script `setup.sh`, under the `kubernetes/samples/charts/util` folder, to install and uninstall Traefik.
+## Install and uninstall the Traefik operator with setupLoadBalancer.sh
+Alternatively, you can run the helper script `setupLoadBalancer.sh`, under the `kubernetes/samples/charts/util` folder, to install and uninstall Traefik.
 
 To install Traefik:
 ```
-$ ./setup.sh create traefik
+$ ./setupLoadBalancer.sh create traefik
```

Comment:
traefik is an ingress controller, not a load balancer, they are different things

Reply:
One of the suggestion by @tbarnes-us to ( Rename to something less generic than 'setup.sh'  ( e.g. setupLoadBalancer.sh ).  Here we are not referring to K8s object rather a generic fronted load balancer to route web request.  Should I rename to setupProxy.sh  or any other suggestion ?

Reply:
Traefik still describes itself as an Ingress Controller, but using a CustomResourceDefinition: https://docs.traefik.io/providers/kubernetes-crd/. It's still confusing to call it a load balancer.